### PR TITLE
ref(similarity): Clean up test layout

### PR DESCRIPTION
**Pull request by tkaemming**
This splits much of the tests into a `MinHashIndexBackendTestMixin` which can be used to test any index backend, regardless of how it is implemented. Each backend will need it's own import/export test since the implementations are backend-specific.

### CODE REVIEW

File: src/sentry/similarity/__init__.py [hunk 1]
```diff
@@ -78,7 +78,7 @@ def _make_index_backend(cluster=None):
             return index
 
     return MetricsWrapper(
-        RedisMinHashIndexBackend(
+        RedisScriptMinHashIndexBackend(
```
Comment (tkaemming) [inline src/sentry/similarity/__init__.py]: Probably didn't need to make this change yet

**Review by tkaemming — COMMENTED** [review]
> Each backend will need it's own import/export test since the implementations are backend-specific.

This is not 100% true, there probably could be a better way to test this

File: src/sentry/similarity/__init__.py (hunk 1 above)
Comment (tkaemming) [reply src/sentry/similarity/__init__.py]: My plan is to add another backend that doesn't depend on Lua scripting and use the base tests to ensure parity between the existing implementation and the new one

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Review by evanpurkhiser — APPROVED** [review]
clean, I like it.

**Pull request merged**